### PR TITLE
Updated profanity manager to only redact in 'public' channels

### PR DESCRIPTION
**Pull request by Uleat**
** Hold for discussion **

### CODE REVIEW

**Comment by Uleat** [comment]
Bumping for conversation..

File: zone/client.cpp [hunk 1]
```diff
@@ -965,16 +964,22 @@ void Client::ChannelMessageReceived(uint8 chan_num, uint8 language, uint8 lang_s
 				}
 			}
 
+			if (EQEmu::ProfanityManager::IsCensorshipActive())
+				EQEmu::ProfanityManager::RedactMessage(message);
+
 			if (!worldserver.SendChannelMessage(this, 0, 4, 0, language, message))
-			Message(0, "Error: World server disconnected");
+				Message(0, "Error: World server disconnected");
```
Comment (dencelle) [inline zone/client.cpp]: this should be using Chat::White instead of 0, new standard

**Comment by Uleat** [comment]
:P 

Yeah, this will definitely need to be updated before a merge.

Conversation-wise, I'd like to see input on the actual principal of redacting speech and what other features would be viable in this implementation.

I'm thinking a client notice (message) upon logging in if the system is active. Kinda like a disclaimer/rule to keep the admin (and us) out of trouble.

**Comment by dencelle** [comment]
that seems like a good idea, live does this in a slightly different way though, instead of redacting the message they "blur" it out (random string of letters that make no sense)... instead of redacting, maybe we should use that method? or maybe when it does get redacted, tell the user that sent the message that it was redacted via a message?

**Comment by Akkadius** [comment]
This has conflicts that need to be settled

**Comment by Akkadius** [comment]
In general, this PR looks fine. I'm not sure why we need to prefix `EQEmu::` with every call to this service

`ProfanityManager` is enough on its own to declare its responsibility, and the whole project is EQEmu scoped

Some conflicts would need to be resolved in order to have this be mergeable

cc @Uleat

**Comment by Akkadius** [comment]
Closing this for now until @Uleat wants to revive this PR

**Pull request closed without merging**